tests/exhaustive-match.test.ts

```diff
@@ -979,4 +979,15 @@ describe('exhaustive()', () => {
       expect(withoutTypo({ sex: 'b', age: 'c' })).toBe(1);
     });
   });
+
+  it('issue #271: P.array should support readonly arrays as its input', () => {
```

Comment:
Nit: I think this would be clearer:
```suggestion
  it('issue #271: P.array should exhaustively match readonly arrays', () => {
```